### ISSUE TITLE
Bug

### ISSUE THREAD

**Issue by jayto** [creagraphy/sb_bt#1]
There is a small bug in your code. The size of a long type is 4 bytes and not just 3. So you have to replace e.g.:

memcpy(&results.Wh, &payload[IDX_VALUE], 3);
by 
memcpy(&results.Wh, &payload[IDX_VALUE], 4);

**Comment by creagraphy** [comment]
Jayto: I might have to check this, but I think the payload only has three bytes for the WattHour value. So if you copy 4 bytes, you will copy a random value extra.

The watt-hour values in my software are correct, so I won't change the code.

**Comment by falco878** [comment]
I experienced the same issue as jayto.

My SB3800's total KWH went negative just past 16777.176 which = 16777176 Wh = FFFFD8 in hex
The sb_bt started logging incorrect numbers, so trouble shooting later with the hex dumps showed
16777215 = FFFFFF is the magic number for 3 byte read.

I modified the code to memcpy(&results.Wh, &payload[IDX_VALUE], 4)and now the values logged
are exactly what is on the SB3800 display.

**Comment by falco878** [comment]
I experienced the same issue as jayto.

My SB3800's total KWH went negative just past 16777.176 which = 16777176 Wh = FFFFD8 in hex
The sb_bt started logging incorrect numbers, so trouble shooting later with the hex dumps showed
16777215 = FFFFFF is the magic number for 3 byte read.

I modified the code to memcpy(&results.Wh, &payload[IDX_VALUE], 4)and now the values logged
are exactly what is on the SB3800 display.
